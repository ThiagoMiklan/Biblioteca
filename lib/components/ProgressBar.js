"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _type_validations = require("../tools/type_validations");

var props_obj = {
  onChange: "function",
  max: "number",
  value: "number",
  definition: "string"
};

var ProgressBar = function ProgressBar(props) {
  (0, _type_validations.validate)(props, props_obj, "ProgressBar");
  var definition = props.definition == undefined ? "progress" : "progress " + props.definition;
  return /*#__PURE__*/_react.default.createElement("progress", {
    onChange: props.onChange,
    className: definition,
    max: props.max,
    value: props.value
  });
};

var _default = ProgressBar;
exports.default = _default;