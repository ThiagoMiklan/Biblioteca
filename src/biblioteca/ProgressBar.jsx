import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../tools/getClassName'


const ProgressBar = (props)=>{
        return (
            <progress className={classnames(getClassName("progress "+ props.definition,"ProgressBar"))}
             max={props.max} 
             value ={props.value}>  
            </progress>
        );
}

ProgressBar.propTypes = {
    definition: PropTypes.string,
    max: PropTypes.string,
    value: PropTypes.string,

}

export default ProgressBar;