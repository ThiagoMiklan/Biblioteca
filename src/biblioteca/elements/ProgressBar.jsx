import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName'


class ProgressBar extends React.Component{


    render(){

        return (
            <progress className={classnames(getClassName(this.props,"ProgressBar"))}
             max={this.props.max} 
             value ={this.props.value}>  
            
            </progress>
        );

    }



}

ProgressBar.propTypes = {
    definition: PropTypes.string,
    max: PropTypes.string,
    value: PropTypes.string,

}

export default ProgressBar;