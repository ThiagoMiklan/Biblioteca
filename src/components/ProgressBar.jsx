// @flow
import React from 'react';
import { validate } from '../tools/type_validations';

type Props = {
    onChange?: ()=> void,
    max?: number,
    value?: number,
    definition?:string
}

const props_obj = {
    onChange: "function",
    max: "number",
    value: "number",
    definition:"string"
}

const ProgressBar = (props: Props)=>{
    validate(props,props_obj,"ProgressBar");
    var definition =  (props.definition == undefined)? "progress": "progress "+ props.definition;
        return (
            <progress onChange={props.onChange} className={definition}
             max={props.max} 
             value ={props.value}>  
            </progress>
        );
}

export default ProgressBar;