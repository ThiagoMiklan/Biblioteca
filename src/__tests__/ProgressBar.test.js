import React from 'react';
import ProgressBar from '../biblioteca/ProgressBar';
import renderer from 'react-test-renderer';

test('progressbar fixed', () => {
    const pb_fixed = renderer.create(
        <>
            <ProgressBar value="15" max="100" />
            <ProgressBar definition="primary" value="15" max="100" />
            <ProgressBar definition="link" value="30" max="100" />
            <ProgressBar definition="info" value="45" max="100" />
            <ProgressBar definition="success" value="60" max="100" />
            <ProgressBar definition="warning" value="75" max="100" />
            <ProgressBar definition="danger" value="90" max="100" />
        </>
    ).toJSON();

    expect(pb_fixed).toMatchSnapshot();

});

test('progressbar sizes', () => {
    const pb_sizes = renderer.create(
        <>
            <ProgressBar definition="small" value="15" max="100" />
            <ProgressBar definition="medium" value="15" max="100" />
            <ProgressBar definition="large" value="15" max="100" />
        </>
    ).toJSON();

    expect(pb_sizes).toMatchSnapshot();

});


test('progressbar inderminate', () => {
    const pb_inderterminated = renderer.create(
        <>
            <ProgressBar definition="small primary" max="100"></ProgressBar>
            <ProgressBar definition="danger" max="100"></ProgressBar>
            <ProgressBar definition="medium dark" max="100"></ProgressBar>
            <ProgressBar definition="large info" max="100"></ProgressBar>
        </>
    ).toJSON();

    expect(pb_inderterminated).toMatchSnapshot();

});
