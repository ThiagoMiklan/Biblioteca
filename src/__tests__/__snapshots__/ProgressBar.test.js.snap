// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progressbar fixed 1`] = `
Array [
  <progress
    className="progress"
    max="100"
    value="15"
  />,
  <progress
    className="progress is-primary"
    max="100"
    value="15"
  />,
  <progress
    className="progress is-link"
    max="100"
    value="30"
  />,
  <progress
    className="progress is-info"
    max="100"
    value="45"
  />,
  <progress
    className="progress"
    max="100"
    value="60"
  />,
  <progress
    className="progress is-warning"
    max="100"
    value="75"
  />,
  <progress
    className="progress is-danger"
    max="100"
    value="90"
  />,
]
`;

exports[`progressbar inderminate 1`] = `
Array [
  <progress
    className="progress is-small is-primary"
    max="100"
  />,
  <progress
    className="progress is-danger"
    max="100"
  />,
  <progress
    className="progress is-medium"
    max="100"
  />,
  <progress
    className="progress is-large is-info"
    max="100"
  />,
]
`;

exports[`progressbar sizes 1`] = `
Array [
  <progress
    className="progress is-small"
    max="100"
    value="15"
  />,
  <progress
    className="progress"
    max="100"
    value="30"
  />,
  <progress
    className="progress is-medium"
    max="100"
    value="45"
  />,
  <progress
    className="progress is-large"
    max="100"
    value="60"
  />,
]
`;
